### PR TITLE
REVSDL-2459 RegisterService request GPB implementation

### PR DESCRIPTION
**Pull request by thinhts**
@bonewell 
@DKlimenko 
@KKarlash 
@VladSemenyuk 
@giangtqh

### CODE REVIEW

**Comment by bonewell** [comment]
Reviewed.

File: src/components/vr_cooperation/src/register_service_request.cc [hunk 1]
```diff
@@ -50,9 +52,26 @@ RegisterServiceRequest::~RegisterServiceRequest() {
 
 void RegisterServiceRequest::Execute() {
   LOG4CXX_AUTO_TRACE(logger_);
+
+  Json::Value params;
+  Json::Reader reader;
+  reader.parse(message_->json_message(), params);
+
+  int result_code = GetHMIResultCode(result_codes::kSuccess);
```
Comment (KKarlash) [inline src/components/vr_cooperation/src/register_service_request.cc]: You get hmi result code to send it to mobile?

File: src/components/vr_cooperation/src/register_service_request.cc [hunk 2]
```diff
@@ -50,9 +52,26 @@ RegisterServiceRequest::~RegisterServiceRequest() {
...
+  bool success = true;
```
Comment (KKarlash) [inline src/components/vr_cooperation/src/register_service_request.cc]: why success is always true?

File: src/components/vr_cooperation/src/register_service_request.cc [hunk 3]
```diff
@@ -50,9 +52,26 @@ RegisterServiceRequest::~RegisterServiceRequest() {
...
+  std::string info;
```
Comment (KKarlash) [inline src/components/vr_cooperation/src/register_service_request.cc]: where will you get info?

File: src/components/vr_cooperation/src/register_service_request.cc (hunk 2 above)
Comment (thinhts) [inline src/components/vr_cooperation/src/register_service_request.cc]: currently I don't know what should I check here. So It's always true.

File: src/components/vr_cooperation/src/register_service_request.cc (hunk 3 above)
Comment (thinhts) [inline src/components/vr_cooperation/src/register_service_request.cc]: currently we don't check anything here so info is empty

**Comment by bonewell** [comment]
I think you have to save information about the registered service in order to handle requests from HMI (ActivateService and ProcessData) in future when you will receive them.

**Comment by thinhts** [comment]
@bonewell  When we receive request from HMI we just need to know what is appid which will be used to send to mobile. 
In ActivateService request, it have appid as an argument. It also save activated appid.
in ProcessData request, we know what appid which was saved by ActivateService request.
So I don't know what information we have to save in RegisterService request

**Comment by bonewell** [comment]
@thinhts Yeah, it seems you are right.

**Comment by thinhts** [comment]
@KKarlash I fix your comment, please review

File: src/components/vr_cooperation/include/vr_cooperation/commands/register_service_request.h [hunk 4]
```diff
@@ -50,7 +51,7 @@ class RegisterServiceRequest : public BaseCommandRequest {
    * @param parent pointer to VRModule
    * @param message Message from mobile
    **/
-  explicit RegisterServiceRequest(
+  RegisterServiceRequest(
       VRModule* parent,
       const application_manager::MessagePtr& message);
```
Comment (giangtqh) [inline src/components/vr_cooperation/include/vr_cooperation/commands/register_service_request.h]: this code can put in 2 lines
```
RegisterServiceRequest(VRModule* parent,
                                      const application_manager::MessagePtr& message)
```

File: src/components/vr_cooperation/src/register_service_request.cc [hunk 5]
```diff
@@ -30,7 +30,9 @@
  POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include "functional_module/function_ids.h"
 #include "vr_cooperation/commands/register_service_request.h"
+#include "vr_cooperation/vr_module_constants.h"
 #include "utils/logger.h"
```
Comment (giangtqh) [inline src/components/vr_cooperation/src/register_service_request.cc]: each section of includes should be sorted
https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

**Comment by giangtqh** [comment]
reviewed

**Comment by KKarlash** [comment]
Reviewed

**Comment by bonewell** [comment]
Reviewed.

**Comment by DKlimenko** [comment]
reviewed

**Pull request merged**